Userland/Libraries/LibJS/Lexer.cpp

```diff
@@ -376,7 +376,7 @@ Optional<u32> Lexer::is_unicode_escape(size_t& identifier_length) const
 {
     GenericLexer lexer(source().substring_view(m_position - 1));
 
-    if (auto code_point_or_error = lexer.consume_escaped_code_point(); !code_point_or_error.is_error()) {
+    if (auto code_point_or_error = lexer.consume_escaped_code_point(false); !code_point_or_error.is_error()) {
```

Comment:
What's the reason not to combine surrogate pairs?

Reply:
It's what the spec says, don't know why it is supposed to happen in some places but not others 🤷.
For reference in IdentifierName https://tc39.es/ecma262/#prod-IdentifierName we can use a UnicodeEscapeSequence https://tc39.es/ecma262/#prod-UnicodeEscapeSequence which can only be `\uXXXX` 4 hex digits or `\u{XXXXXX}` some hex digits.

Whereas ~string~ regex has https://tc39.es/ecma262/#prod-RegExpUnicodeEscapeSequence RegExpUnicodeEscapeSequence which does allow for combining surrogate pairs.

String is a little harder but has to do with source text https://tc39.es/ecma262/#sec-static-semantics-sv I think?

All in all a comment would be good here so I will add that 👍